compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/cyclefind/GlobalVariableRefAnalyzer.java

```diff
@@ -192,9 +192,7 @@ private boolean isGlobalVarSymbol(BSymbol symbol) {
             return false;
         }
 
-        return ((symbol.tag & SymTag.VARIABLE) == SymTag.VARIABLE) ||
-                ((symbol.tag & SymTag.CONSTANT) == SymTag.CONSTANT);
```

Comment:
what is the reason to remove the condition on CONSTANT here?

Reply:
A lock doesnt have to depend on a constant.